### DISCUSSION TITLE
Copying ports

### DISCUSSION THREAD

**Discussion by mikemowery** [netbox-community/netbox#7540] [Q&A]
We have many of the same devices but they are equipped differently with many types of cards. The actual device stays the same but the interfaces change. We built the device as a simple device with no interfaces. When you find another device with the exact same interfaces is there some way to copy everything over and just change the name? Cloning is great but it only clones the template device and not the interfaces. I'm looking to copy everything and just change the device name so I'm not adding all the interfaces. I hope this make sense. Thank in advance......

**Comment by bogdancordos** [comment]
Cloning a device is using the _Device Type_ template. I'm not certain if there is a way to copy only the  interfaces over, however, I will suggest you create a _Device Type_ for every variance you have.

**Comment by PieterL75** [comment]
You could create a Custom Script that does that for you... 
with 2 parameters : Source Device and Destination Device
And then a simple loop that copies the interfaces of one device to the other

**Reply by mikemowery** [reply]
Thank you for this idea. This sounds great but I'm not a coder. Not sure how that would look or be incorporated into Netbox?

**Reply by kkthxbye-code** [reply]
Here's an example of a simple script to copy all interfaces between two existing devices. So create the source device with all interfaces, clone that device like normal and select both in the script and run it.

```python
from extras.scripts import Script, ObjectVar
from dcim.models import Device


class CopyInterfaces(Script):
    class Meta:
        name = "Copy Interfaces"

    source_device = ObjectVar(model=Device, label="Source Device", required=True)
    target_device = ObjectVar(model=Device, label="Target Device", required=True)

    def run(self, data, commit):
        source_device = data["source_device"]
        target_device = data["target_device"]

        for interface in source_device.interfaces.all():
            if not target_device.interfaces.filter(name=interface.name).exists():
                interface.pk = None
                interface.device = target_device
                interface.save()
```

**Reply by DanSheps** [reply]
Remember to call clean() before save() as you could run into data integrity issues if you don't.
